src/parse/parser.rs

```diff
@@ -40,6 +39,61 @@ pub enum ParseResult {
     ValuesDone,
 }
 
+#[derive(Debug)]
+#[doc(hidden)]
```

Comment:
Do we really need to make it `pub #[doc(hidden)]`?

Reply:
I had trouble using this in `src/build/app/mod.rs` without `pub` here.

Reply:
Then `pub(crate)` (no hidden) will do just well

Reply:
What about it?

Reply:
It is actually being used in a public interface `get_matches_with`. Cannot make this private. I have removed the `doc(hidden)` though.

Reply:
Removing "hidden" is not a good idea since this struct in definitely not a part of public API. 

`Parser` (the struct `get_matches_with` defined in) is not public either; it's just `pub hidden` for some reason I don't understand. Actually there are many such items in clap and, for my inexperienced eye, they don't have to be. 

Let's just leave it hidden for now, I'll look at these visibility modifiers tomorrow (really tomorrow).